### PR TITLE
ci: enable cloud provider E2E tests

### LINKED ISSUE #2427: Introduce Cloud Provider E2E tests
Extend the continuos-delivery workflow to allow running E2E tests on Cloud Providers (EKS, AKS, GKE).
The Cloud provider tests are disabled by default, and will only run in case the following GH variables are found inside the target repository:

* vars.AKS_ENABLED is set to 'true'
* vars.EKS_ENABLED is set to 'true'
* vars.GKE_ENABLED is set to 'true'

3 new JSON files (`aks_versions`, `eks_versions` and `gke_versions`) will hold each Cloud Provider version on which E2E tests will be executed. These files will add up to the existing `kind_versions`, and will be used by `e2e-matrix-generator.py` to build a matrix of E2E test jobs.

### PR DESCRIPTION
**Pull request by NiccoloFei**
Closes #2427

### CODE REVIEW

**Review by litaocdl — APPROVED** [review]

File: .github/gke_versions.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+[
```
Comment (jsilvela) [inline .github/gke_versions.json:1]: in the CNP repo I see 1.27 at the top of the gke_versions file.
Noting to double-check....

File: hack/e2e/eks-cluster.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+apiVersion: eksctl.io/v1alpha5
```
Comment (jsilvela) [inline hack/e2e/eks-cluster.yaml:1]: I think this file needs to be called `hack/e2e/eks-cluster.yaml.template`.
Checking...

Comment (jsilvela) [reply hack/e2e/eks-cluster.yaml:1]: Confirmed. Fixing.

File: .github/workflows/kind-k8s-versions-check.yml [hunk 3]
```diff
@@ -1,54 +0,0 @@
-# Checks the latest kind image version from `https://registry.hub.docker.com/v2/repositories/kindest/node/tags`,
```
Comment (jsilvela) [inline .github/workflows/kind-k8s-versions-check.yml:1]: not sure why we're deleting this workflow. There's a public cloud versions check in CNP that wer're not bringing in instead. Wondering if we forgot to.

**Review by jsilvela — APPROVED** [review]

**Review by mnencia — APPROVED** [review]

**Pull request merged**